ACPs/77-reinventing-subnets/README.md

```diff
@@ -46,77 +46,160 @@ For standalone networks like the Avalanche Primary Network, this is done by conn
+```text
++--------------+----------+-----------+
+|      codecID :   uint16 |   2 bytes |
++--------------+----------+-----------+
+|       typeID :   uint32 |   4 bytes |
++--------------+----------+-----------+
+|     subnetID : [32]byte |  32 bytes |
++--------------+----------+-----------+
+|       nodeID : [32]byte |  32 bytes |
++--------------+----------+-----------+
+|       weight :   uint64 |   8 bytes |
++--------------+----------+-----------+
+| blsPublicKey : [48]byte |  48 bytes |
++--------------+----------+-----------+
+|       expiry :   uint64 |   8 bytes |
++--------------+----------+-----------+
+                          | 134 bytes |
+                          +-----------+
+```
 
-This transaction is, by design, not required to be submitted by the validator themselves. With the Ed25519 signature, the validator guarantees that they can only be added to the validator set if `Signer` corresponds to the `blsPublicKey` and `Balance` >= `balance`. The `RegisterSubnetValidatorTx` is considered invalid if those two properties are not satisfied.
+- `codecID` is the codec version used to serialize the payload and is hardcoded to `0x0000`
+- `typeID` is the payload type identifier and is `0x00000000` for this transaction
+- `subnetID`, `nodeID`, `weight`, and `blsPublicKey` are for the Subnet Validator being added
+- `expiry` is the time after which this message is invalid. After the P-Chain timestamp is past `expiry`, this Avalanche Warp Message can no longer be used to add the `nodeID` to the validator set of `subnetID`.
 
-For a `RegisterSubnetValidatorTx` to be valid, `Balance` must be >= the greater of 5 $AVAX or two weeks of the current fee. This prevents Subnet Validators from being added with too low of a `Balance` where they become immediately delinquent based on the continuous fee mechanism defined below. A Subnet Validator can leave at any time before the initial $AVAX is consumed and claim the remaining `Balance` to the `ChangeOwner` defined in the transaction.
+    `validationID` is defined as the SHA256 hash of the `Payload` of the `AddressedCall`. This SHA256 hash will be used for replay protection. Used `validationID`s will be stored on the P-Chain. If a `RegisterSubnetValidatorTx`'s `validationID` has already been used, the transaction will be considered invalid. To prevent storing an unbounded number of `validationID`s, the `expiry` is required to be no longer than 48 hours in the future of the time the transaction is issued on the P-Chain. Any `validationIDs` with `expiry` more than 48 hours in the past can be flushed from the P-Chain's state.
```

Comment:
Why do we wait 48 hours past the `expiry` to flush the `validationID`?

Reply:
for replay protection. If we flush it 24 hours after, somebody can reissue the same message corresponding to the `validationID` and it would be valid since `expiry` is 24 hours in the future.

Reply:
I'm not really understanding. If we flush it 24 hours after the expiry, the expiry is 24 hours in the past, not in the future.

Reply:
Ah I understand the confusion- this was poorly worded on my part.
```suggestion
    `validationID` is defined as the SHA256 hash of the `Payload` of the `AddressedCall`. This SHA256 hash will be used for replay protection. Used `validationID`s will be stored on the P-Chain. If a `RegisterSubnetValidatorTx`'s `validationID` has already been used, the transaction will be considered invalid. To prevent storing an unbounded number of `validationID`s, the `expiry` is required to be no longer than 48 hours in the future of the time the transaction is issued on the P-Chain. Any `validationIDs` with an `expiry` in the past can be flushed from the P-Chain's state.
```